### PR TITLE
General purpose applications

### PR DESCRIPTION
**Pull request by tanismar**
Given that all .sh scripts use the same modules to be run for every demo, we (Kammo and me) though it would be better, for further uses and new users, to have a general application (.xml), instead of a new application for each demo. So we created one for simulation and one for real robot, called icubDemoScriptSIM and icubDemoScript, respectively

### CODE REVIEW

**Comment by alecive** [comment]
@tanismar by default, `xml` application templates should not define a specific node to run the modules on, but rather use some dummy placeholders such that `pwrNode1`, `console` etc . Can you please address this issue on the two `xml` files you have created? I cannot do it on my own because only you have access to your fork.

File: app/scripts/icubDemoScript.xml.template [hunk 1]
```diff
@@ -1,63 +1,63 @@
...
-        <parameters>--autoStart</parameters>
-        <node>localhost</node>
+        <parameters>--autoStart --robot icub</parameters>
+        <node>icub15</node>
         <stdio></stdio>
     </module>
     <module>
         <name>iSpeak</name>
-        <parameters>--package acapelacmd --package_options icub_ita</parameters>
+        <parameters>--package acapelacmd --robot icub --package_options icub_ita</parameters>
```
Comment (alecive) [inline app/scripts/icubDemoScript.xml.template]: wasn't this different, i.e. `acapela-tts` ?

Comment (alecive) [inline app/scripts/icubDemoScript.xml.template]: @tanismar

File: app/scripts/icubDemoScriptSIM.xml.template [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    </module>
+    <module>
+        <name>iCubBlinker</name>
+        <parameters>--autoStart --robot icubSim</parameters>
+        <node>localhost</node>
+        <stdio></stdio>
+    </module>
+    <module>
+        <name>iSpeak</name>
+        <parameters>--package acapelacmd --robot icubSim --package_options icub_ita</parameters>
```
Comment (alecive) [inline app/scripts/icubDemoScriptSIM.xml.template]: @tanismar  also here you should change to `acapela-tts` . Further, `--robot` flag is not needed (`iSpeak` is robot-agnostic).

Comment (tanismar) [inline app/scripts/icubDemoScriptSIM.xml.template]: ciao Ale, 
scussa per il dissaggio, ho cambiato il host dell'App e corretto la chiamata a Acapella. 

Spero addesso sia tutto a posto. 

Grazie a te!
Tanis
________________________________________
De: Alessandro Roncone [notifications@github.com]
Enviado: lunes, 20 de julio de 2015 13:03
Para: robotology/funny-things
CC: Tanis Mar
Asunto: Re: [funny-things] General purpose applications (#5)

In app/scripts/icubDemoScriptSIM.xml.template<https://github.com/robotology/funny-things/pull/5#discussion_r34984482>:

> +    </module>
> +    <module>
> +        <name>ctpService</name>
> +        <parameters>--robot icubSim --part torso</parameters>
> +        <node>localhost</node>
> +        <stdio></stdio>
> +    </module>
> +    <module>
> +        <name>iCubBlinker</name>
> +        <parameters>--autoStart --robot icubSim</parameters>
> +        <node>localhost</node>
> +        <stdio></stdio>
> +    </module>
> +    <module>
> +        <name>iSpeak</name>
> +        <parameters>--package acapelacmd --robot icubSim --package_options icub_ita</parameters>


@tanismar<https://github.com/tanismar> also here you should change to acapela-tts . Further, --robot flag is not needed (iSpeak is robot-agnostic).

�

**Pull request merged**